### ISSUE TITLE
64 bit Windows builds are broken according to ci.narc.ro

### ISSUE THREAD

**Issue by BorkBorkGoesTheCode** [CleverRaven/Cataclysm-DDA#21802]
Starting with number 6737

**Comment by ZhilkinSerg** [comment]
http://ci.narc.ro/job/Cataclysm-Matrix/Graphics=Tiles,Platform=Windows_x64/lastBuild/console

````
08:47:47 src/map.cpp: In member function 'bool map::has_floor_or_support(const tripoint&) const':
08:47:47 src/map.cpp:2154:6: error: assuming signed overflow does not occur when assuming that (X - c) > X is always false [-Werror=strict-overflow]
08:47:47  bool map::has_floor_or_support( const tripoint &p ) const
08:47:47       ^
08:47:47 src/map.cpp:2154:6: error: assuming signed overflow does not occur when assuming that (X - c) > X is always false [-Werror=strict-overflow]
08:47:47 cc1plus: all warnings being treated as errors
08:47:47 Makefile:703: recipe for target 'objwin/tiles/map.o' failed
08:47:47 make: *** [objwin/tiles/map.o] Error 1
08:47:47 make: *** Waiting for unfinished jobs....
08:47:48 Build step 'Conditional step (single)' marked build as failure
08:47:48 SSH: Current build result is [FAILURE], not going to run.
08:47:48 Recording fingerprints
08:47:48 IRC notifier plugin: Sending notification to: #CDDA-dev
08:47:48 Finished: FAILURE
````

**Comment by ZhilkinSerg** [comment]
# OS X
http://ci.narc.ro/job/Cataclysm-Matrix/Graphics=Curses,Platform=OSX/lastBuild/console

````
08:30:55 In file included from src/projectile.cpp:1:
08:30:55 In file included from src/projectile.h:5:
08:30:55 src/damage.h:74:29: error: implicit instantiation of undefined template 'std::__1::array<int, 10>'
08:30:55     std::array<int, NUM_DT> dealt_dams;
08:30:55                             ^
08:30:55 /home/narc/osxcross/osxcross/target/bin/../SDK/MacOSX10.11.sdk/usr/include/c++/v1/__tuple:95:65: note: template is declared here
08:30:55 template <class _Tp, size_t _Size> struct _LIBCPP_TYPE_VIS_ONLY array;
08:30:55                                                                 ^
08:30:55 In file included from src/projectile.cpp:1:
08:30:55 In file included from src/projectile.h:5:
08:30:55 src/damage.h:84:31: error: implicit instantiation of undefined template 'std::__1::array<float, 10>'
08:30:55     std::array<float, NUM_DT> resist_vals;
08:30:55                               ^
08:30:55 /home/narc/osxcross/osxcross/target/bin/../SDK/MacOSX10.11.sdk/usr/include/c++/v1/__tuple:95:65: note: template is declared here
08:30:55 template <class _Tp, size_t _Size> struct _LIBCPP_TYPE_VIS_ONLY array;
08:30:55                                                                 ^
08:30:55 2 errors generated.
08:30:55 Makefile:703: recipe for target 'obj/projectile.o' failed
08:30:55 make: *** [obj/projectile.o] Error 1
08:30:55 make: *** Waiting for unfinished jobs....
08:31:02 Build step 'Conditional step (single)' marked build as failure
08:31:02 SSH: Current build result is [FAILURE], not going to run.
08:31:02 Recording fingerprints
08:31:02 IRC notifier plugin: Sending notification to: #CDDA-dev
08:31:02 Finished: FAILURE
````

**Comment by PlasmaChroma** [comment]
Got my build to complete by adding this to the start of map::valid_move

`assert(to.z > std::numeric_limits<int>::min());`

**Comment by bpwatts** [comment]
Also have this error on Mac OS X:

clang++  -DRELEASE -DMACOSX -DGIT_VERSION -DTILES -ffast-math -Os  -Wall -Wextra    -fsigned-char -stdlib=libc++ -Werror -std=c++11 -MMD -MP -mmacosx-version-min=10.7 -F/Library/Frameworks -I/Library/Frameworks/SDL2.framework/Headers -I/Library/Frameworks/SDL2_image.framework/Headers -I/Library/Frameworks/SDL2_ttf.framework/Headers -c src/bonuses.cpp -o obj/tiles/bonuses.o
In file included from src/bonuses.cpp:2:
src/damage.h:74:29: error: implicit instantiation of undefined template 'std::__1::array<int, 10>'
    std::array<int, NUM_DT> dealt_dams;
                            ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__tuple:116:65: note: template is declared here
template <class _Tp, size_t _Size> struct _LIBCPP_TYPE_VIS_ONLY array;
                                                                ^
In file included from src/bonuses.cpp:2:
src/damage.h:84:31: error: implicit instantiation of undefined template 'std::__1::array<float, 10>'
    std::array<float, NUM_DT> resist_vals;
                              ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__tuple:116:65: note: template is declared here
template <class _Tp, size_t _Size> struct _LIBCPP_TYPE_VIS_ONLY array;
                                                                ^
2 errors generated.

This after a make clean with git pull && make NATIVE=osx OSX_MIN=10.7 RELEASE=1 LOCALIZE=0 CLANG=1 FRAMEWORK=1 TILES=1 as the runline

**Comment by shazow** [comment]
I get the same error as @bpwatts on current master (commit 70487bd7a87cfe15ece5d6568be41a151e102022), my build command is `make NATIVE=osx OSX_MIN=10.7 RELEASE=1 TILES=1 LOCALIZE=0 USE_HOME_DIR=1 CLANG=1 SOUND=1`

Switching to a random-ish commit from last week (11f550d8144b5e3cc19e122d2a802c0fc50fb21b), it builds fine.

**Comment by bpwatts** [comment]
How do you switch to another commit and then switch back? I always manage to flub it up.

**Comment by shazow** [comment]
`git checkout yourcommithere` and then to switch back `git checkout master`

**Comment by bpwatts** [comment]
Oh hell, I was using some --reset hard thing. Thanks for the training!

**Comment by BrianLefler** [comment]
OSX issue seems like it's missing the include for the array template, which can be fixed by adding this at the top of damage.h: #include <array>

Will submit a fix in a sec as soon as I've finished compiling.

**Comment by BorkBorkGoesTheCode** [comment]
Fix #21810 merged. Will close if it works.

**Comment by BorkBorkGoesTheCode** [comment]
OSX issue fixed. Windows 64 still a problem

**Comment by BorkBorkGoesTheCode** [comment]
Fix candidate: #21856
